pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -47,8 +49,10 @@ const (
 )
 
 var (
-	cfg = config.New()
-	log = logf.Log.WithName(controllerName)
+	cfg                = config.New()
+	log                = logf.Log.WithName(controllerName)
+	hasCheckedEnmasse  = false
+	canWatchEnmasseVal = false
```

Comment:
Are these global vars used anywhere apart from inside the `canWatchEnmasse` function?

Reply:
They are not.  It would probably be more golike to wrap the canWatchEnmasse in its own pacake to isolate those variables.

Reply:
What if they were just declared at the top of the function? Or does the pre-existing state matter (i.e. what it may have set them to last time it was called)?

Reply:
The idea is that we only need to make that check once since it needs a brief timeout window.  That timeout, however, is blocking so these variable make it check only once
 at startup.